tests/e2e/test_internal_trades.py

```diff
@@ -0,0 +1,188 @@
+    return sum(
+        a.from_amount for a in internal_trade_list
+        if a.from_token == Address(token_str)
+    )
+
+
+def get_internal_token_transfers(
+        dune: DuneAnalytics,
+        tx_hash: str,
+        period_start: datetime,
+        period_end: datetime
+) -> list[InternalTokenTransfer]:
+    path = "./queries/slippage"
+    slippage_subquery = File(path=path,
+                             name="subquery_batchwise_internal_transfers.sql")
+    select_transfers_file = File(path=path, name="select_in_out_with_buffers.sql")
+    query = "\n".join(
+        [
+            dune.open_query(slippage_subquery.filename()),
+            dune.open_query(select_transfers_file.filename())
+        ]
+    )
+    data_set = dune.fetch(
+        query_str=query,
+        network=Network.MAINNET,
+        name='Internal Token Transfer Accounting',
+        parameters=[
+            QueryParameter.text_type('TxHash', tx_hash),
+            QueryParameter.date_type("StartTime", period_start),
+            QueryParameter.date_type("EndTime", period_end),
+        ])
+    return [
+        InternalTokenTransfer.from_dict(row) for row in data_set
+    ]
+
+
+class TestDuneAnalytics(unittest.TestCase):
+    def setUp(self):
+        self.dune_connection = DuneAnalytics.new_from_environment()
+        self.period_start = datetime.strptime('2022-03-01', "%Y-%m-%d")
+        self.period_end = datetime.strptime('2022-03-12', "%Y-%m-%d")
+
+    def test_one_buffer_trade(self):
+        """
+        tx: 0xd6b85ada980d10a11a5b6989c72e0232015ce16e7331524b38180b85f1aea6c8
+        This transaction has 1 buffer trades wNXM for Ether
+        """
+        internal_token_transfers = get_internal_token_transfers(
+            dune=self.dune_connection,
+            tx_hash='0xd6b85ada980d10a11a5b6989c72e0232015ce16e7331524b38180b85f1aea6c8',
+            period_start=self.period_start,
+            period_end=self.period_end,
+        )
+        internal_buffer_trades = [
+            a for a in internal_token_transfers
+            if a.transfer_type == 'INTERNAL_TRADE'
+        ]
+        self.assertEqual(len(internal_buffer_trades), 1 * 2)
+        # We had 0.91 USDT positive slippage
+        self.assertEqual(
+            token_slippage(
+                '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+                internal_token_transfers
+            ), 916698)
+
+    def test_zero_buffer_trade(self):
+        """
+        tx: 0x31ab7acdadc65944a3f9507793ba9c3c58a1add35de338aa840ac951a24dc5bc
+        Although there are two quite similar trades TOKE for ETH,
+        the script should **not** detect a buffer trade
+        """
+        internal_token_transfers = get_internal_token_transfers(
+            dune=self.dune_connection,
+            tx_hash='0x31ab7acdadc65944a3f9507793ba9c3c58a1add35de338aa840ac951a24dc5bc',
+            period_start=self.period_start,
+            period_end=self.period_end,
+        )
+        internal_buffer_trades = [
```

Comment:
Maybe we could write a function for that. These 3 lines repeat so frequently

Reply:
Yes I'm already thinking about it. There is a todo for it in the code.

Reply:
I think you may have highlighted the wrong three lines, but I suspect you are referring to the filtering for internal trades. Which we can definitely remove. Also we should make the transfer type and enum.